features/springframework-security/src/main/java/org/opennms/web/springframework/security/RequestHeaderPreAuthenticationProcessingFilter.java

```diff
@@ -65,6 +76,26 @@ public void afterPropertiesSet() {
         } else {
             LOG.info("Request header pre-authentication filter is disabled.");
         }
+        setAuthenticationDetailsSource(new AuthenticationDetailsSource());
+    }
+
+    private class AuthenticationDetailsSource extends WebAuthenticationDetailsSource  {
+        @Override
+        public WebAuthenticationDetails buildDetails(HttpServletRequest context) {
+            return new PreAuthenticatedGrantedAuthoritiesWebAuthenticationDetails(context, getGrantedAuthorities(context));
+        }
+
+        private Collection<? extends GrantedAuthority> getGrantedAuthorities(HttpServletRequest context) {
+            String roles = "";
+            if (m_authoritiesHeader != null) {
+                roles = context.getHeader(m_authoritiesHeader);
+            }
+            if (StringUtils.isBlank(roles)) {
+                return Collections.emptyList();
+            }
+            return Arrays.stream(roles.split(","))
+                    .map(SimpleGrantedAuthority::new).collect(Collectors.toList());
```

Comment:
is it worth doing a `trim()` also in case any spaces slip through?

Reply:
Good point. Will add.

Reply:
And a filter for empty strings, too?